File: SMarketPlace/SMarketPlace.Web/Modules/Sucursales/TblSucursalDidi/TblSucursalDidiRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace SMarketPlace.Sucursales;

[LookupScript(Expiration = -1)]
[ConnectionKey("Default"), Module("Sucursales"), TableName("TBLSucursalDidi")]
[DisplayName("Sucursal Didi"), InstanceName("Sucursal Didi")]
[ReadPermission(PermissionKeys.Didi)]
[ModifyPermission(PermissionKeys.Didi)]
public sealed class TblSucursalDidiRow : Row<TblSucursalDidiRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Local Sap"), Column("LocalSAP"), Size(4), IdProperty, QuickSearch, NameProperty, ForeignKey("[dbo].[tblSucursal]", "LocalSap"), LeftJoin("jSucursales"), TextualField("NombreSuc")]
    [LookupEditor(typeof(TblSucursalRow))]
    public string LocalSap
    {
        get => fields.LocalSap[this];
        set => fields.LocalSap[this] = value;
    }
    [DisplayName("Nombre de Sucursal"), Expression("jSucursales.[NombreSuc]"), QuickSearch]
    [LookupEditor(typeof(TblSucursalRow))]
    public string NombreSucursal
    {
        get => fields.NombreSucursal[this];
        set => fields.NombreSucursal[this] = value;
    }
    public class RowFields : RowFieldsBase
    {
        public StringField LocalSap;
        public StringField NombreSucursal;
    }
    public bool TrackWithChecks { get; internal set; }
}

File: SMarketPlace/SMarketPlace.Web/Modules/ServerTypes/Sucursales/TblSucursalDidiRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface TblSucursalDidiRow {
    LocalSap?: string;
    NombreSucursal?: string;
}

export abstract class TblSucursalDidiRow {
    static readonly idProperty = 'LocalSap';
    static readonly nameProperty = 'LocalSap';
    static readonly localTextPrefix = 'Sucursales.TblSucursalDidi';
    static readonly lookupKey = 'Sucursales.TblSucursalDidi';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TblSucursalDidiRow>('Sucursales.TblSucursalDidi') }
    static async getLookupAsync() { return getLookupAsync<TblSucursalDidiRow>('Sucursales.TblSucursalDidi') }

    static readonly deletePermission = 'Sucursales:Didi';
    static readonly insertPermission = 'Sucursales:Didi';
    static readonly readPermission = 'Sucursales:Didi';
    static readonly updatePermission = 'Sucursales:Didi';

    static readonly Fields = fieldsProxy<TblSucursalDidiRow>();
}